Sources/LiveKit/Support/ConnectivityListener.swift

```diff
@@ -0,0 +1,70 @@
+import Network
+
+internal protocol ConnectivityListenerDelegate: AnyObject {
+
+    func connectivityListener(_: ConnectivityListener, didUpdate hasConnectivity: Bool)
+    // network remains to have connectivity but path changed
+    func connectivityListener(_: ConnectivityListener, didSwitch path: NWPath)
+}
+
+internal extension ConnectivityListenerDelegate {
+    func connectivityListener(_: ConnectivityListener, didUpdate hasConnectivity: Bool) {}
+    func connectivityListener(_: ConnectivityListener, didSwitch path: NWPath) {}
+}
+
+internal extension NWPath {
+
+    func hasConnectivity() -> Bool {
+        if case .satisfied = status { return true }
+        return false
+    }
+}
+
+internal class ConnectivityListener: MulticastDelegate<ConnectivityListenerDelegate> {
+
+    static let shared = ConnectivityListener()
+
+    public private(set) var hasConnectivity: Bool {
+        didSet {
+            guard oldValue != hasConnectivity else { return }
+            notify { $0.connectivityListener(self, didUpdate: self.hasConnectivity) }
+        }
+    }
+
+    public private(set) var path: NWPath {
+        didSet {
+            defer { self.hasConnectivity = path.hasConnectivity() }
+
+            if oldValue.hasConnectivity(), path.hasConnectivity(),
+               // first interface changed
+               (oldValue.availableInterfaces.first != path.availableInterfaces.first)
+                // same interface but gateways changed (detect wifi network switch)
+                || (oldValue.gateways != path.gateways) {
```

Comment:
Is this a `route` change? When does this happen?

Is it possible to detect IP address change? I guess if it is NATted it is not possible. To keep it simple, I guess letting any IP address change take the ICE failed route is reasonable.